src/ConfiguredMediator.php

```diff
@@ -30,23 +34,49 @@ public function __construct()
 
     public function uninstall(Composer $composer, IOInterface $io)
     {
-        $this->composer = $composer;
-        $this->io       = $io;
+        parent::uninstall($composer, $io);
         $this->removePhars();
     }
 
     public function installOrUpdateFunction(PackageEvent $event): void
     {
+        $gnuPG = $this->createGnuPG();
+        // we do not want to crash if no GnuPG was found
+        // but display a noticeable warning to the user
+        if ($gnuPG === null) {
+            $this->getIO()->write(
+                PHP_EOL .
+                '    <warning>WARNING</warning>' . PHP_EOL .
+                '    No GPG installation found! Use installed PHARs with care. ' . PHP_EOL .
+                '    Consider installing GnuPG to verify PHAR authenticity.' . PHP_EOL .
+                '    If you need help installing GnuPG visit http://phar.io/install-gnupg' . PHP_EOL
+            );
+        }
+
         $installer = $this->createInstallerFromConfig($this->config, $event);
-        $installer->install($this->config->phars());
+        $installer->install(
+            $this->config->phars(),
+            $this->config->keyDirectory() ? $this->createKeyDirectory($this->config) : null,
+            $gnuPG
+        );
+    }
+
+    public function createGnuPG(): ?GnuPG
```

Comment:
I hate these `nullable` thingies... 

Can't we have a 'MissingGnuPg`- class?

Reply:
You mean to throw an exception?

Reply:
Either that, or returnsomething that behaves like the thing that you would otherwise return. Like a class that will trigger a warning when calling `verify` but otherwise returns true.

That way we can leave out the null-checks in the calling code and the warnings are put into an appropruate place

Reply:
I hear what you are saying but I think in order to make the error message visible and to not disrupt the actual install output we should keep the "ugly" null checks and let the plugin class trigger the warning at the appropriate time over the appropriate place. I know not perfekt but looks much cleaner for the user.